Makefile

```diff
@@ -12,11 +13,22 @@ clean:
 # the "-tags daemon" part is temporary
 .PHONY: test
 test:
-	./scripts/test/unit $(shell go list ./... | grep -v /vendor/)
+	go test -tags daemon -v ${PACKAGES}
 
 .PHONY: test-coverage
 test-coverage:
-	./scripts/test/unit-with-coverage
+	( for pkg in ${PACKAGES}; do \
+		go test -tags daemon \
+			-cover \
+			-coverprofile=profile.out \
+			-covermode=atomic \
+			$$pkg || exit;\
+		\
+		if test -f profile.out; then \
+			cat profile.out >> coverage.txt; \
+			rm profile.out; \
+		fi; \
+	done )
```

Comment:
When would you actually benefit from `make` not running the coverage because no files are changed? Any test target seems naturally `.PHONY` to me.

I'm also not a fan of leaving a bunch of garbage `.out` files around. Removing them after the concat is the right thing to do.

This solution makes it much harder to transition to something other than a makefile, does less than the current bash script, and is significantly harder to read (in my opinion).

I don't think it's unreasonable to say that most maintainers of this repo are more familiar with bash than they are Makefile syntax, so maintaining bash is going to be a better option (until there is a better go tool that does this properly).